third_party/move/testing-infra/transactional-test-runner/src/vm_test_harness.rs

```diff
@@ -329,37 +335,73 @@ impl<'a> MoveTestAdapter<'a> for SimpleVMTestAdapter<'a> {
     }
 }
 
-pub fn format_vm_error(e: &VMError, comparison_mode: bool) -> String {
+pub fn format_vm_error(e: &VMError, comparison_mode: bool, verbose: bool) -> String {
     let location_string = match e.location() {
         Location::Undefined => "undefined".to_owned(),
         Location::Script => "script".to_owned(),
         Location::Module(id) => format!("0x{}::{}", id.address().short_str_lossless(), id.name()),
     };
-    format!(
-        "{{
+    if verbose {
+        let message_str = match &e.message() {
+            Some(message_str) => message_str,
+            None => "None",
+        };
+        format!(
+            "{{
     major_status: {major_status:?},
     sub_status: {sub_status:?},
     location: {location_string},
     indices: {indices},
     offsets: {offsets},
+    message: {message},
+    exec_state: {exec_state:?},
 }}",
-        major_status = e.major_status(),
-        sub_status = e.sub_status(),
-        location_string = location_string,
-        // TODO maybe include source map info?
-        indices = if comparison_mode {
-            // During comparison testing, abstract this data.
-            "redacted".to_string()
-        } else {
-            format!("{:?}", e.indices())
-        },
-        offsets = if comparison_mode {
-            // During comparison testing, abstract this data.
-            "redacted".to_string()
-        } else {
-            format!("{:?}", e.offsets())
-        },
-    )
+            major_status = e.major_status(),
```

Comment:
Is it possible to reduce duplicated code by concatenating normal error and the verbose message?